### PR TITLE
R2: Support using legacy editors with new validation components

### PR DESCRIPTION
**Pull request by const314**

### CODE REVIEW

File: packages/interim/src/index.ts [hunk 1]
```diff
@@ -2,10 +2,11 @@ import registerComponent from './core/component_registrator';
 import { compileGetter } from './core/utils/data';
 import type { ItemLike } from './ui/collection/ui.collection_widget.base';
 import ValidationEngine from './ui/validation_engine';
+import DefaultAdapter from './ui/validation/default_adapter';
 
 export type { ValidationRule, ValidationRuleType, CompareRule, CustomRule, EmailRule, NumericRule, PatternRule, RangeRule, RequiredRule, StringLengthRule } from './ui/validation_rules';
 export type { ValidationResult } from './ui/validator';
-export { registerComponent, ItemLike, compileGetter, ValidationEngine };
+export { registerComponent, ItemLike, compileGetter, ValidationEngine, DefaultAdapter };
```
Comment (MikeVitik) [inline packages/interim/src/index.ts]: rename it to a more specific ValidationAdapter.

Comment (const314) [reply packages/interim/src/index.ts]: renamed to EditorValidationAdapter

**Review by IlyaKhD — APPROVED** [review]

File: packages/react/src/components/validation/components/legacy-editor-connector.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const legacyEditor = (renderer(domAccessor.current?.parentNode as Element) as any).data('dx-validation-target');
+    const editorAdapter = new EditorValidationAdapter(legacyEditor, validator);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    editorAdapter.getName = () => (legacyEditor as any).option('name');
+    setEditorAdapter(editorAdapter);
+  });
+
+  return <span ref={domAccessor} style={{ display: 'none' }} />;
```
Comment (MikeVitik) [inline packages/react/src/components/validation/components/legacy-editor-connector.tsx]: try to use "display: contents"

**Review by MikeVitik — APPROVED** [review]

File: packages/react/src/components/validation/components/legacy-editor-connector.tsx (hunk 2 above)
Comment (const314) [reply packages/react/src/components/validation/components/legacy-editor-connector.tsx]: fixed

**Pull request merged**